tools/test_utils/fullservice.py

```diff
@@ -0,0 +1,215 @@
+        
+
+
+    def start(self):
+        cmd = ' '.join([
+            'mkdir -p /tmp/wallet-db/',
+            f'&& {constants.FULLSERVICE_DIR}/target/release/full-service',
+            '--wallet-db /tmp/wallet-db/wallet.db',
+            '--ledger-db /tmp/ledger-db/',
+            '--peer insecure-mc://localhost:3200',
+            '--peer insecure-mc://localhost:3201',
+            f'--tx-source-url file://{constants.MOBILECOIN_DIR}/target/release/mc-local-network/node-ledger-distribution-0',
+            f'--tx-source-url file://{constants.MOBILECOIN_DIR}/target/release/mc-local-network/node-ledger-distribution-1',
+        ])
+        print('===================================================')
+        print('starting full service')
+        print(cmd)
+        self.full_service_process = subprocess.Popen(cmd, shell=True)
+
+    def stop(self):
+        try:
+            subprocess.check_output("pkill full-service", shell=True)
+        except subprocess.CalledProcessError as exc:
+            if exc.returncode != 1:
+                raise
+
+    # return the result field of the request
+    def _request(self, request_data):
+        self.request_count += 1
+        print('sending request to full service')
+        url = 'http://127.0.0.1:9090/wallet'
+        default_params = {
+            "jsonrpc": "2.0",
+            "api_version": "2",
+            "id": self.request_count,
+        }
+        request_data = {**request_data, **default_params}
+
+        print(f'request data: {request_data}')
+
+        try:
+            parsed_url = urlparse(url)
+            connection = http.client.HTTPConnection(parsed_url.netloc)
+            connection.request('POST', parsed_url.path, json.dumps(request_data), {'Content-Type': 'application/json'})
+            r = connection.getresponse()
+        except ConnectionError:
+            raise ConnectionError
+
+        raw_response = None
+        try:
+            raw_response = r.read()
+            response_data = json.loads(raw_response)
+        except ValueError:
+            raise ValueError('API returned invalid JSON:', raw_response)
+        print(f'request returned {response_data}')
+
+        # TODO requests might be flakey due to timing issues... waiting 2 seconds to bypass most of these issues
+        time.sleep(2)
+        if 'error' in response_data:
+            return response_data['error']
+        else:
+            return response_data['result']
+
+    def import_account(self, mnemonic) -> bool:
+        print(f'importing full service account {mnemonic}')
+        params = {
+            'mnemonic': mnemonic,
+            'key_derivation_version': '2',
+        }
+        r = self._request({
+            "method": "import_account",
+            "params": params
+        })
+
+        if 'error' in r:
+            if 'Diesel Error: UNIQUE constraint failed' in r['error']['data']['details']:
+                return True
+            else:
+                return False
```

Comment:
This can be shortened and be documented.
```suggestion
            # If we failed due to a unique constraint, it means the account already exists
            return 'Diesel Error: UNIQUE constraint failed' in r['error']['data']['details']